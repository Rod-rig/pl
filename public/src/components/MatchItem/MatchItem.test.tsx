import * as React from "react";
import { MemoryRouter } from "react-router-dom";
import * as renderer from "react-test-renderer";
import { MatchItem } from "./";

describe("MatchItem", () => {
  const wrapMemoryRouter = (comp: React.ReactNode) => {
    return <MemoryRouter>{comp}</MemoryRouter>;
  };

  const predictedWrongMatch = wrapMemoryRouter(
    <MatchItem
      awayTeam="Test1"
      awayScore={1}
      awayLogo="logo1"
      homeTeam="Test2"
      homeScore={2}
      homeLogo="logo2"
      id="test"
      status={0}
    />,
  );
  const futureMatch = wrapMemoryRouter(
    <MatchItem awayTeam="Test1" homeTeam="Test2" id="test" />,
  );
  const playedMatch = wrapMemoryRouter(
    <MatchItem
      awayTeam="Test1"
      awayScore={1}
      awayLogo="logo1"
      homeTeam="Test2"
      homeScore={2}
      homeLogo="logo2"
      id="test"
      status={1}
    />,
  );

  it("should render predicted wrong match correctly", () => {
    const tree = renderer.create(predictedWrongMatch).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render future match correctly", () => {
    const tree = renderer.create(futureMatch).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render played match correctly", () => {
    const tree = renderer.create(playedMatch).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
