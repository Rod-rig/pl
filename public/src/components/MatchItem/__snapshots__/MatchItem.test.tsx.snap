// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MatchItem should render future match correctly 1`] = `
<div
  aria-disabled={false}
  className="MuiButtonBase-root MuiListItem-root Component-matchItem-9 MuiListItem-gutters MuiListItem-divider MuiListItem-button"
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="button"
  tabIndex={0}
>
  <div
    className="MuiListItemText-root Component-text-12 Component-home-6"
  >
    <div>
      Test2
    </div>
    <div
      className="MuiListItemIcon-root Component-homeIcon-7 Component-root-34"
    >
      <div
        className="MuiAvatar-root Component-logo-33"
      >
        <img
          className="MuiAvatar-img"
          src="https://cdn2.iconfinder.com/data/icons/sport-items-2/512/football_soccer_badge_shield_ball_heraldic-128.png"
        />
      </div>
    </div>
  </div>
  <div
    className="Component-score-11"
  >
    <div
      className="Component-dash-2"
    >
      :
    </div>
  </div>
  <div
    className="MuiListItemText-root Component-text-12 Component-guest-5"
  >
    <div
      className="MuiListItemIcon-root Component-awayIcon-1 Component-root-34"
    >
      <div
        className="MuiAvatar-root Component-logo-33"
      >
        <img
          className="MuiAvatar-img"
          src="https://cdn2.iconfinder.com/data/icons/sport-items-2/512/football_soccer_badge_shield_ball_heraldic-128.png"
        />
      </div>
    </div>
    <div>
      Test1
    </div>
  </div>
</div>
`;

exports[`MatchItem should render played match correctly 1`] = `
<div
  aria-disabled={false}
  className="MuiButtonBase-root MuiListItem-root Component-matchItem-9 MuiListItem-gutters MuiListItem-divider MuiListItem-button"
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="button"
  tabIndex={0}
>
  <div
    className="MuiListItemText-root Component-text-12 Component-home-6"
  >
    <div>
      Test2
    </div>
    <div
      className="MuiListItemIcon-root Component-homeIcon-7 Component-root-34"
    >
      <div
        className="MuiAvatar-root Component-logo-33"
      >
        <img
          className="MuiAvatar-img"
          src="https://cdn2.iconfinder.com/data/icons/sport-items-2/512/football_soccer_badge_shield_ball_heraldic-128.png"
        />
      </div>
    </div>
  </div>
  <div
    className="Component-score-11 Component-green-4"
  >
    <div>
      2
    </div>
    <div
      className="Component-dash-2"
    >
      :
    </div>
    <div>
      1
    </div>
  </div>
  <div
    className="MuiListItemText-root Component-text-12 Component-guest-5"
  >
    <div
      className="MuiListItemIcon-root Component-awayIcon-1 Component-root-34"
    >
      <div
        className="MuiAvatar-root Component-logo-33"
      >
        <img
          className="MuiAvatar-img"
          src="https://cdn2.iconfinder.com/data/icons/sport-items-2/512/football_soccer_badge_shield_ball_heraldic-128.png"
        />
      </div>
    </div>
    <div>
      Test1
    </div>
  </div>
</div>
`;

exports[`MatchItem should render predicted wrong match correctly 1`] = `
<div
  aria-disabled={false}
  className="MuiButtonBase-root MuiListItem-root Component-matchItem-9 MuiListItem-gutters MuiListItem-divider MuiListItem-button"
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="button"
  tabIndex={0}
>
  <div
    className="MuiListItemText-root Component-text-12 Component-home-6"
  >
    <div>
      Test2
    </div>
    <div
      className="MuiListItemIcon-root Component-homeIcon-7 Component-root-34"
    >
      <div
        className="MuiAvatar-root Component-logo-33"
      >
        <img
          className="MuiAvatar-img"
          src="https://cdn2.iconfinder.com/data/icons/sport-items-2/512/football_soccer_badge_shield_ball_heraldic-128.png"
        />
      </div>
    </div>
  </div>
  <div
    className="Component-score-11 Component-red-10"
  >
    <div>
      2
    </div>
    <div
      className="Component-dash-2"
    >
      :
    </div>
    <div>
      1
    </div>
  </div>
  <div
    className="MuiListItemText-root Component-text-12 Component-guest-5"
  >
    <div
      className="MuiListItemIcon-root Component-awayIcon-1 Component-root-34"
    >
      <div
        className="MuiAvatar-root Component-logo-33"
      >
        <img
          className="MuiAvatar-img"
          src="https://cdn2.iconfinder.com/data/icons/sport-items-2/512/football_soccer_badge_shield_ball_heraldic-128.png"
        />
      </div>
    </div>
    <div>
      Test1
    </div>
  </div>
</div>
`;
